test/src/loaders/workspace_uploader/workspace_uploader_test.py

```diff
@@ -1,5 +1,210 @@
+    tmp_dir, assembly_dirs, assembly_names, _, workspace_id, upload_env_key, _ = setup_and_teardown
+    collection_source_dir = Path(tmp_dir) / "collectionssource"
+    upload_file_ext = ["genomic.fna.gz"]
+
+    count, wait_to_upload_assemblies = workspace_uploader._fetch_assemblies_to_upload(
+        upload_env_key, workspace_id, collection_source_dir, upload_file_ext
+    )
+    
+    expected_count = len(assembly_names)
+    expected_wait_to_upload_assemblies = {
+        assembly_name: assembly_dir
+        for assembly_name, assembly_dir in zip(assembly_names, assembly_dirs)
+    }
+
+    assert expected_count == count
+    assert expected_wait_to_upload_assemblies == wait_to_upload_assemblies
+
+
+def test_prepare_skd_job_dir_to_upload(setup_and_teardown):
+    tmp_dir, assembly_dirs, assembly_names, _, _, _, _ = setup_and_teardown
+    wait_to_upload_assemblies = {
+        assembly_name: assembly_dir
+        for assembly_name, assembly_dir in zip(assembly_names, assembly_dirs)
+    }
+
+    conf = Mock()
+    username = "kbase"
+    job_dir = loader_helper.make_job_dir(tmp_dir, username)
+    conf.job_data_dir = loader_helper.make_job_data_dir(job_dir)
+    data_dir = workspace_uploader._prepare_skd_job_dir_to_upload(conf, wait_to_upload_assemblies)
+    
+    assert sorted(os.listdir(data_dir)) == sorted(assembly_names)
+
+
+def test_post_process(setup_and_teardown):
+    tmp_dir, assembly_dirs, assembly_names, upas, workspace_id, upload_env_key, _ = setup_and_teardown
+    upload_dir = Path(tmp_dir) / "upload_dir"
+    upload_dir.mkdir()
+    output_dir = Path(tmp_dir) / "output_dir"
+    output_dir.mkdir()
+
+    upa = upas[0]
+    host_assembly_dir = assembly_dirs[0]
+    assembly_name = assembly_names[0]
+
+    workspace_uploader._post_process(
+        upload_env_key,
+        workspace_id,
+        host_assembly_dir,
+        assembly_name,
+        upload_dir,
+        output_dir,
+        upa,
+    )
+
+    data, uploaded = workspace_uploader._read_upload_status_yaml_file(
+        upload_env_key, workspace_id, host_assembly_dir, assembly_name
+    )
+    expected_data = {
+        upload_env_key: {workspace_id: {"file_name": assembly_name, "upa": upa}}
+    }
+
+    assert uploaded
+    assert expected_data == data
+    assert os.readlink(os.path.join(upload_dir, upa)) == os.path.join(output_dir, upa)
+
+
+def test_upload_assembly_to_workspace(setup_and_teardown):
+    _, _, assembly_names, upas, workspace_id, _, _  = setup_and_teardown
+    upa = upas[0]
+    assembly_name = assembly_names[0]
+    file_path = "/path/to/file/in/AssembilyUtil"
+
+    conf = Mock()
+    conf.asu.save_assembly_from_fasta2.return_value = {"upa": upa}
+    upa_res = workspace_uploader._upload_assembly_to_workspace(
+        conf, workspace_id, file_path, assembly_name
+    )
+
+    expected_upa = upa.replace("/", "_")
+    assert expected_upa == upa_res
+    assert conf.asu.save_assembly_from_fasta2.call_count == 1
```

Comment:
Why not check the arguments are as expected?

Reply:
Updated!

Reply:
Sorry, I'm not seeing the change for this one? It looks like it's still just testing the call count, not the arguments. You'd need to do something like
```python
conf.asu.save_assembly_from_fasta2.assert_called_once_with(<args go here>)
```

Reply:
Updated!

Reply:
:+1: